File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque

def solution(arr):
    answer = []
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    
    for i in range(len(arr)):
        if i == 0:
            answer.append(arr[i])
        
        if arr[i] != answer[-1]:
            answer.append(arr[i])
    
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public List<Integer> solution(int []arr) {
        int[] answer = {};
        int lastNum = -1;
        List<Integer> returnData = new ArrayList<>();
        for(int i = 0 ; i < arr.length; i++){
            if(arr[i] != lastNum){
                returnData.add(arr[i]);
            }
            lastNum = arr[i];
        }

        return returnData;
    }
}